tests/remove_ribo_rna.nf.test

```diff
@@ -946,31 +40,16 @@ nextflow_pipeline {
         }
 
         then {
+            def path_from_outdir   = UTILS.getAllFilesFromDir(params.outdir, null, UTILS.exclusionRegexesForUnstableFileNames)
+            def path_snapshottable = UTILS.getAllFilesFromDir(params.outdir, UTILS.snapshottablePatterns, UTILS.exclusionRegexesForUnstableFileContents)
+
             assertAll(
-                { assert workflow.success },
+                { assert workflow.success},
                 { assert snapshot(
+                    workflow.trace.tasks().size(),
                     UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
-                    file("${params.outdir}/custom/out/genome_transcriptome.fasta").name,
-                    file("${params.outdir}/custom/out/genome_transcriptome.gtf").name,
-                    file("${params.outdir}/fastqc/raw/RAP1_IAA_30M_REP1_raw.html").name,
-                    file("${params.outdir}/fastqc/raw/RAP1_IAA_30M_REP1_raw.zip").name,
-                    file("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP1_raw.html").name,
-                    file("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP1_raw.zip").name,
-                    file("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP2_raw.html").name,
-                    file("${params.outdir}/fastqc/raw/RAP1_UNINDUCED_REP2_raw.zip").name,
-                    file("${params.outdir}/fastqc/raw/WT_REP1_raw.html").name,
-                    file("${params.outdir}/fastqc/raw/WT_REP1_raw.zip").name,
-                    file("${params.outdir}/fastqc/raw/WT_REP2_raw.html").name,
-                    file("${params.outdir}/fastqc/raw/WT_REP2_raw.zip").name,
-                    file("${params.outdir}/multiqc/star_salmon/multiqc_report.html").name,
-                    file("${params.outdir}/trimgalore/RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP1_trimmed_1.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP1_trimmed_2.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP2_trimmed_1.fastq.gz_trimming_report.txt").name,
-                    file("${params.outdir}/trimgalore/WT_REP2_trimmed_2.fastq.gz_trimming_report.txt").name
+                    path_from_outdir.collect{ file -> path_snapshottable.contains(file) ? null : file.getName()}.findAll{ it },
```

Comment:
What's going on here? We don't need to limit this to non-shapshottable files, the complement can contain all files, except the ones with dates in their names.

Reply:
I don't want to snapshot the filenames of the file that I have already snapshotted, so I remove them from this list.

Reply:
I think we should just retain the complete list of files (without the ones with variable names) as something to check, it's simpler. And actually, doing that we don't need to check the number of files.